### PR TITLE
Move role attribute in “<pre>: The Preformatted Text element”

### LINKED ISSUE #12361: Issue with "<pre>: The Preformatted Text element": w3c Validator throwing error on <figure> example
MDN URL: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/pre

#### What information was incorrect, unhelpful, or incomplete?
When I checked the code on the w3c validator. It throws an error "A figure element with a figcaption descendant must not have a role attribute."

#### Specific section or headline?
Validator error on adding role="img" on <figure>

#### Did you test this? If so, how?
By adding this code on Validator. Please let me know how to fix this and if we need to add role="img"

<details>
<summary>MDN Content page report details</summary>

* Folder: `en-us/web/html/element/pre`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/pre
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/html/element/pre/index.md
* Last commit: https://github.com/mdn/content/commit/e752a8fe731fbdb9061e39a6e2b797a7ccb7e2b5
* Document last modified: 2022-01-19T17:57:58.000Z

</details>

### PR DESCRIPTION
**Pull request by himanshugarg**
#### Summary
The role attribute was misplaced in the example. And the example reads ascii art chars which it is not supposed to. Verified that it reads more sensibly now:

Voice | Screenreader
--- | ---
 `graphic` ascii cow | NVDA on Windows 11
 `caption` a cow saying ... preformatted text ... | -
 `caption` characters | -
ascii cow `graphic` | Google Talkback on Android 11
 a cow saying ... characters | -
 ascii cow `image` | Orca on Debian 11
 a cow saying ... characters `caption` | -

#### Motivation
| I want to publicly demonstrate my own skills to improve my chances of getting a job.

#### Supporting details
Thanks @9Nish4 for reporting
Thanks @scottaohara for the review and the link https://w3c.github.io/html-aria/#example-8

#### Related issues
Fixes #12361

#### Metadata
- [x] Fixes a typo, bug, or other error

### CODE REVIEW

File: files/en-us/web/html/element/pre/index.md [hunk 1]
```diff
@@ -90,12 +90,13 @@ It is important to provide an alternate description for any images or diagrams c
 
 People experiencing low vision conditions and browsing with the aid of assistive technology such as a screen reader may not understand what the preformatted text characters are representing when they are read out in sequence.
 
-A combination of the {{HTMLElement("figure")}} and {{HTMLElement("figcaption")}} elements, supplemented by a combination of an {{htmlattrxref("id")}} and the [ARIA](/en-US/docs/Web/Accessibility/ARIA) `role` and [`aria-labelledby`](/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-labelledby) attributes allow the preformatted text to be announced as an image, with the `figcaption` serving as the image's alternate description.
+A combination of the {{HTMLElement("figure")}} and {{HTMLElement("figcaption")}} elements, supplemented by a combination of an {{htmlattrxref("id")}}
+and [`aria-labelledby`](/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-labelledby) attributes allow the preformatted text to be announced as an image, with the `figcaption` serving as the image's alternate description.
```
Comment (sideshowbarker) [inline files/en-us/web/html/element/pre/index.md:94]:
```suggestion
A combination of the {{HTMLElement("figure")}} and {{HTMLElement("figcaption")}} elements, supplemented by a combination of {{htmlattrxref("id")}}
and [`aria-labelledby`](/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-labelledby) attributes, allows the preformatted text to be announced as an image, with the `figcaption` serving as the image's alternate description.
```

File: files/en-us/web/html/element/pre/index.md [hunk 2]
```diff
@@ -90,12 +90,13 @@ It is important to provide an alternate description for any images or diagrams c
 
 People experiencing low vision conditions and browsing with the aid of assistive technology such as a screen reader may not understand what the preformatted text characters are representing when they are read out in sequence.
 
-A combination of the {{HTMLElement("figure")}} and {{HTMLElement("figcaption")}} elements, supplemented by a combination of an {{htmlattrxref("id")}} and the [ARIA](/en-US/docs/Web/Accessibility/ARIA) `role` and [`aria-labelledby`](/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-labelledby) attributes allow the preformatted text to be announced as an image, with the `figcaption` serving as the image's alternate description.
+A combination of the {{HTMLElement("figure")}} and {{HTMLElement("figcaption")}} elements, supplemented by a combination of {{htmlattrxref("id")}}
```
Comment (sideshowbarker) [inline files/en-us/web/html/element/pre/index.md:93]:
```suggestion
A combination of {{HTMLElement("figure")}} and {{HTMLElement("figcaption")}} elements, supplemented by a combination of {{htmlattrxref("id")}}
```

File: files/en-us/web/html/element/pre/index.md [hunk 3]
```diff
@@ -90,12 +90,13 @@ It is important to provide an alternate description for any images or diagrams c
 
 People experiencing low vision conditions and browsing with the aid of assistive technology such as a screen reader may not understand what the preformatted text characters are representing when they are read out in sequence.
 
-A combination of the {{HTMLElement("figure")}} and {{HTMLElement("figcaption")}} elements, supplemented by a combination of an {{htmlattrxref("id")}} and the [ARIA](/en-US/docs/Web/Accessibility/ARIA) `role` and [`aria-labelledby`](/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-labelledby) attributes allow the preformatted text to be announced as an image, with the `figcaption` serving as the image's alternate description.
+A combination of {{HTMLElement("figure")}} and {{HTMLElement("figcaption")}} elements, supplemented by a combination of {{htmlattrxref("id")}}
+and [`aria-labelledby`](/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-labelledby) attributes, allows the preformatted text to be announced as an image, with the `figcaption` serving as the image's alternate description.
```
Comment (scottaohara) [inline files/en-us/web/html/element/pre/index.md:94]: removing the `role=img` makes this line not make sense anymore.

"allows the preformatted text to be announced as an image, with the `figcaption` serving as the image's alternate description."

without the `role=img` there is no longer an "image".

Rather this should be revised further to resemble [this example from ARIA in HTML](https://w3c.github.io/html-aria/#example-8).

**Review by scottaohara — COMMENTED** [review]
this seems more like it was a bug with talkback.  removing the `role=img` means that further updates should be made here.

File: files/en-us/web/html/element/pre/index.md [hunk 4]
```diff
@@ -90,13 +90,13 @@ It is important to provide an alternate description for any images or diagrams c
 
 People experiencing low vision conditions and browsing with the aid of assistive technology such as a screen reader may not understand what the preformatted text characters are representing when they are read out in sequence.
 
-A combination of the {{HTMLElement("figure")}} and {{HTMLElement("figcaption")}} elements, supplemented by a combination of an {{htmlattrxref("id")}} and the [ARIA](/en-US/docs/Web/Accessibility/ARIA) `role` and [`aria-labelledby`](/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-labelledby) attributes allow the preformatted text to be announced as an image, with the `figcaption` serving as the image's alternate description.
+A combination of the {{HTMLElement("figure")}} and {{HTMLElement("figcaption")}} elements, supplemented by the [ARIA](/en-US/docs/Web/Accessibility/ARIA) `role` and [`aria-label`](/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-label) attributes on the `pre` element allow the preformatted text to be announced as an image, with the `figcaption` serving as the image's alternate description.
```
Comment (scottaohara) [inline files/en-us/web/html/element/pre/index.md:93]:
```suggestion
A combination of the {{HTMLElement("figure")}} and {{HTMLElement("figcaption")}} elements, supplemented by the [ARIA](/en-US/docs/Web/Accessibility/ARIA) `role` and [`aria-label`](/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-label) attributes on the `pre` element allow the preformatted ASCII art to be announced as an image with alternative text, and the `figcaption` serving as the image's caption.
```

**Comment by sideshowbarker** [comment]
@scottaohara If you are now ready to approve this, I think we could merge it

**Comment by scottaohara** [comment]
i don't have approve permissions, but looks good to me @sideshowbarker

**Review by sideshowbarker — APPROVED** [review]

**Pull request merged**